backend/open_webui/utils/usage.py

```
@@ -0,0 +1,216 @@
+
+    def run(self, response: Union[dict, bytes]) -> None:
+        if not isinstance(response, (dict, bytes)):
+            logger.warning("[credit_deduct] response is type of %s", type(response))
+            return
+        # prompt messages
+        messages = self.body.get("messages", [])
+        if not messages:
+            raise HTTPException(status_code=400, detail="prompt messages is empty")
+        # stream
+        if self.is_stream:
+            _response = response.decode("utf-8").strip().lstrip("data: ")
```

Ruff (0.8.2)

Using `.strip()` with multi-character strings is misleading

(B005)